### ISSUE TITLE
JarHell failure for :x-pack:qa:smoke-test-plugins:integTest

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#66971]
**Build scan**: https://gradle-enterprise.elastic.co/s/bymk3bcfpznta

**Repro line**:
./gradlew :x-pack:qa:smoke-test-plugins:integTest -Dtests.fips.enabled=true

**Reproduces locally?**:
Yes

**Applicable branches**:
7.x

**Failure history**:

Somehow it just started failing. Also it does not fail for `master` for even `7.11` (tried locally)

**Failure excerpt**:
```
| Exception in thread "main" java.lang.IllegalStateException: failed to load plugin ingest-attachment due to jar hell
| 	at org.elasticsearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:622)
| 	at org.elasticsearch.plugins.InstallPluginCommand.jarHellCheck(InstallPluginCommand.java:855)
| 	at org.elasticsearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:823)
| 	at org.elasticsearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:868)
| 	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:256)
| 	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:226)
| 	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
| 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:127)
| 	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:91)
| 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:127)
| 	at org.elasticsearch.cli.Command.main(Command.java:90)
| 	at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:47)
| Caused by: java.lang.IllegalStateException: jar hell!
| class: org.bouncycastle.LICENSE
| jar1: /dev/shm/elastic+elasticsearch+7.x+matrix-java-periodic-fips/ES_RUNTIME_JAVA/openjdk14/nodes/general-purpose/x-pack/qa/smoke-test-plugins/build/testclusters/integTest-0/distro/7.12.0-DEFAULT/plugins/.installing-2490609617693076849/bcprov-jdk15on-1.64.jar
| jar2: /dev/shm/elastic+elasticsearch+7.x+matrix-java-periodic-fips/ES_RUNTIME_JAVA/openjdk14/nodes/general-purpose/x-pack/qa/smoke-test-plugins/build/testclusters/integTest-0/distro/7.12.0-DEFAULT/lib/bc-fips-1.0.2.jar
| 	at org.elasticsearch.bootstrap.JarHell.checkClass(JarHell.java:287)
| 	at org.elasticsearch.bootstrap.JarHell.checkJarHell(JarHell.java:195)
| 	at org.elasticsearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:620)
| 	... 11 more
```

**Comment by tvernum** [comment]
Ingest attachment isn't supported on FIPS, so the failure itself isn't surprising.
What I need to track down is why it works on `7.11` and `master` but not `7.x`

**Comment by ywangd** [comment]
> Ingest attachment isn't supported on FIPS, so the failure itself isn't surprising.
> What I need to track down is why it works on `7.11` and `master` but not `7.x`

The master branch actually has [logic](https://github.com/elastic/elasticsearch/blob/master/x-pack/qa/smoke-test-plugins/build.gradle#L15) to filter this plugin out. So it makes sense. But I am not sure why 7.11 does not fail. digging more ...

**Comment by tvernum** [comment]
It looks like 7.x has been broken since a6fce75a1e596d5b09610e18fc0fa627efa10920

**Comment by tvernum** [comment]
That commit is for PR #65807 which was `master` and `7.x` only, which explains why 7.11 works.

**Comment by ywangd** [comment]
> That commit is for PR #65807 which was `master` and `7.x` only, which explains why 7.11 works.

I am not sure exactly what this change does. But if it is not applied to 7.11, shouldn't 7.11 fail more broadly with other type of failures?

**Comment by tvernum** [comment]
> shouldn't 7.11 fail more broadly with other type of failures?

I _think_ (and @jkakavas will keep me honest) that this PR did the reverse - it fixed some parts of the build that weren't actually running in FIPS mode so that they did run in FIPS mode.
So it means that some things in 7.11 might not be being tested 100% correctly.

Which means, since that change, this test has been running in FIPS mode (before that it was not) so the failure is a result of fixing the environment to correctly cover the scenario we intended, but when it was backported to 7.x we missed some of the consequences of that (because this build.gradle was out of sync between 7.x and master)

**Comment by ywangd** [comment]
Oh no. It does fail for 7.11 on CI: https://gradle-enterprise.elastic.co/s/yx3vaje73qx5i

**Comment by tvernum** [comment]
Resolved by #66977 and #67062
